### PR TITLE
Deprecate vendor branches with `knife cookbook site install`

### PR DESCRIPTION
**Pull request by coderanger**
So we can kill it for Chef 14. This adds a CLI option to not vendor and warns if you don't use it. I think I did the option correctly but I should probably make some kind of test for this.

💣 💥

### CODE REVIEW

File: lib/chef/knife/cookbook_site_install.rb [hunk 1]
```diff
@@ -90,7 +102,7 @@ def run
         upstream_file = File.join(@install_path, "#{@cookbook_name}.tar.gz")
 
         @repo.sanity_check
```
Comment (lamont-granquist) [inline lib/chef/knife/cookbook_site_install.rb]: should this also be protected by `unless config[:no_vendor]`?  and possibly `@repo` shouldn't be built unless we're vendoring (although scope...)

**Comment by tas50** [comment]
@coderanger Care to rebase this one?

**Comment by tas50** [comment]
There's some test failures here @coderanger, but it would be nice to get this one in before the deprecation window closes.

**Comment by lamont-granquist** [comment]
unit tests are actually failing 'cuz new deprecation warnings get thrown

**Comment by tas50** [comment]
@lamont @coderanger Any reason we don't just warn now and yank this in 15? Seems a lot simpler than requiring folks to put it in the configs to keep it until we ship the next major.

**Comment by lamont** [comment]
You probably wanted @lamont-granquist

**Comment by tas50** [comment]
Closing in favor of #7012

**Pull request closed without merging**